site/pages/cookies.tsx

```diff
@@ -0,0 +1,144 @@
+                                        value="on"
+                                        defaultChecked={trackingDefaultValue === "on"}
+                                    />
+                                    <label
+                                        className="govuk-label govuk-radios__label"
+                                        htmlFor="accept-analytics-cookies"
+                                    >
+                                        Use cookies that measure my website use
+                                    </label>
+                                </div>
+                                <div className="govuk-radios__item">
+                                    <input
+                                        className="govuk-radios__input"
+                                        type="radio"
+                                        id="decline-analytics-cookies"
+                                        name="tracking"
+                                        value="off"
+                                        defaultChecked={trackingDefaultValue === "off"}
+                                    />
+                                    <label
+                                        className="govuk-label govuk-radios__label"
+                                        htmlFor="decline-analytics-cookies"
+                                    >
+                                        Do not use cookies that measure my website use
+                                    </label>
+                                </div>
+                            </div>
+                        </fieldset>
+                    </div>
+                    <div>
+                        <h2 className="govuk-heading-m">Strictly necessary cookies</h2>
+                        <p className="govuk-body">These essential cookies are used to remember your progress.</p>
+                        <p className="govuk-body">They always need to be on.</p>
+                        <p className="govuk-link govuk-!-font-size-19">
+                            <a className="govuk-link govuk-!-font-size-19" href="/cookieDetails">
+                                Find out more about our usage of cookies in our cookie policy
+                            </a>
+                        </p>
+                    </div>
+                    <button
+                        className="govuk-button mt-4"
+                        data-module="govuk-button"
+                        id="save-cookie-preferences-button"
+                    >
+                        Save Changes
+                    </button>
+                </>
+            </CsrfForm>
+        </div>
+    </TwoThirdsLayout>
+);
+
+export const getServerSideProps = (ctx: NextPageContextWithSession): { props: CookiePreferencesProps } => {
+    const csrfToken = getCsrfToken(ctx);
+    const cookies = parseCookies(ctx);
+
+    const settingsSaved = (ctx.query?.settingsSaved ?? "false") === "true";
```

Comment:
Curious if we need the `?? false` - if `ctx.query?.settingsSaved` equates to `undefined`, then that value is falsy.

Reply:
refactored

Reply:
@lj-kpmg just to confirm I'm not giving poor advice here

Reply:
`!!ctx.query?.settingsSaved` will equate to true if settingsSaved is "false" since it's a string and "false" is truthy, `ctx.query?.settingsSaved === "true"` should work although the original way is fine if that's erroring for some reason

Reply:
corrected